pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -573,91 +587,96 @@ def newpkgcb(pkgId, name, arch):
+                                    dc.content.is_modular = True
+                                    for dc_modulemd in nevra_to_module[dc.content.nevra]:
+                                        dc.extra_data['modulemd_relation'].append(dc_modulemd)
+                                        dc_modulemd.extra_data['package_relation'].append(dc)
+
+                                if dc.content.name in pkgname_to_groups.keys():
+                                    for dc_group in pkgname_to_groups[dc.content.name]:
+                                        dc.extra_data['group_relations'].append(dc_group)
+                                        dc_group.extra_data['related_packages'].append(dc)
+
+                                await self.put(dc)
 
                     elif results[0].url == updateinfo_url:
                         updateinfo_xml_path = results[0].path
                         metadata_pb.increment()
 
                         updates = await RpmFirstStage.parse_updateinfo(updateinfo_xml_path)
 
-                        advisories_pb.total = len(updates)
+                        advisories_pb_total = len(updates)
+                        advisories_pb.total = advisories_pb_total
                         advisories_pb.state = 'running'
                         advisories_pb.save()
 
-                        for update in updates:
-                            update_record = UpdateRecord(**UpdateRecord.createrepo_to_dict(update))
-                            update_record.digest = hash_update_record(update)
-                            future_relations = {'collections': defaultdict(list), 'references': []}
-
-                            for collection in update.collections:
-                                coll_dict = UpdateCollection.createrepo_to_dict(collection)
-                                coll = UpdateCollection(**coll_dict)
-
-                                for package in collection.packages:
-                                    pkg_dict = UpdateCollectionPackage.createrepo_to_dict(package)
-                                    pkg = UpdateCollectionPackage(**pkg_dict)
-                                    future_relations['collections'][coll].append(pkg)
-
-                            for reference in update.references:
-                                reference_dict = UpdateReference.createrepo_to_dict(reference)
-                                ref = UpdateReference(**reference_dict)
-                                future_relations['references'].append(ref)
-
-                            advisories_pb.increment()
-                            dc = DeclarativeContent(content=update_record)
-                            dc.extra_data = future_relations
-                            await self.put(dc)
+                        for batch in grouper(updates, 250):
+                            advisories_pb.increase_by(len(batch))
+
+                            for update in batch:
```

Comment:
Hard to tell from the diff but the only change is the above 3 lines and the removal of .increment() below